chainer/optimizer.py

```diff
@@ -554,9 +599,11 @@ def reallocate_cleared_grads(self):
                     xp = cuda.get_array_module(param.data)
                     param.grad = xp.zeros_like(param.data)
 
-    def call_hooks(self):
+    def call_hooks(self, hooks=None):
```

Comment:
I noticed that by adding the `hooks` option, users can invoke hooks that are not registered. I think it does not break documented behaviors of this modules but would like to know if it is an intended behavior.

Reply:
Good catch. I did not intend this behavior. I think it's best to avoid this.

Reply:
One idea is to feed `str` that takes either `pre` or `post` and get hooks inside of the method, it loses flexibility, though.